src/ser/mod.rs

```diff
@@ -476,3 +476,38 @@ impl Error {
             "top-level serializer supports only maps and structs".into())
     }
 }
+
+#[cfg(test)]
+mod test {
+
+    use super::*;
```

Comment:
But moving it to its own integration test works too. Could you just rename the file to `test_serialize.rs`?